polyfill/lib/absolute.mjs

```diff
@@ -1,126 +1,197 @@
 import { ES } from './ecmascript.mjs';
 import { EPOCHNANOSECONDS, CreateSlots, GetSlot, SetSlot } from './slots.mjs';
+import { absolute as STRING } from './regex.mjs';
 
-export function Absolute(epochNanoseconds) {
-  if (!(this instanceof Absolute)) return new Absolute(epochNanoseconds);
-  CreateSlots(this);
-  SetSlot(this, EPOCHNANOSECONDS, epochNanoseconds);
-}
+export class Absolute {
+  constructor(epochNanoseconds) {
+    CreateSlots(this);
+    SetSlot(this, EPOCHNANOSECONDS, epochNanoseconds);
+  }
+
+  getEpochSeconds() {
+    const value = GetSlot(this, EPOCHNANOSECONDS);
+    const epochSeconds = Math[value.ms < 0 ? 'ceil' : 'floor'](value.ms / 1000);
+    return epochSeconds;
+  }
+  getEpochMilliseconds() {
+    const value = GetSlot(this, EPOCHNANOSECONDS);
+    const epochMilliSeconds = value.ms;
+    return epochMilliSeconds;
+  }
+  getEpochMicroseconds() {
+    const value = GetSlot(this, EPOCHNANOSECONDS);
+    const epochNanoseconds = BigInt(value.ms) * BigInt(1e6) + (BigInt(value.ns) % BigInt(1e6));
+    return epochNanoseconds / BigInt(1e3);
+  }
+  getEpochNanoseconds() {
+    const value = GetSlot(this, EPOCHNANOSECONDS);
+    const epochMicroseconds = BigInt(value.ms) * BigInt(1e6) + (BigInt(value.ns) % BigInt(1e6));
+    return epochMicroseconds;
+  }
+
+  plus(durationLike = {}) {
+    const duration = ES.GetIntrinsic('%Temporal.duration%')(durationLike);
```

Comment:
Is there any way these functions could be referred to by the from method instead? This name looks vestigial.

Reply:
Sort of, it should be moved to an algorithm which happens to do the same thing as `Duration.from`. I don't want patching of `Duration` to change the behaviour of `plus` though. So actually calling `Duration.from` seems wrong.